src/services/plugin.service.ts

```diff
@@ -12,6 +13,37 @@ export class PluginService {
     if (this.plugins[pluginName]) {
       throw new Error(`${pluginName} already exists as a plugin.`);
     }
-    return (this.plugins[pluginName] = new PluginLoader(pluginName, args) as IPlugin);
+
+    const reference = (this.plugins[pluginName] = new PluginLoader(pluginName, args) as IPlugin);
+
+    this.registerAliases(pluginName);
+
+    return reference;
+  }
+
+  registerAliases(pluginName: string, _args?: any): void {
+    const aliases = this.plugins[pluginName].pluginAlias;
+
+    if (aliases === undefined) {
+      this.aliases[pluginName] = pluginName;
+    } else {
+      if (!aliases.includes(pluginName)) {
+        if (this.aliases[pluginName] !== undefined) {
+          throw new Error(
+            `Duplicate alias detected: ${pluginName} is claiming its primary alias ${pluginName}, prevouisly claimed by ${this.aliases[pluginName]}.`
+          );
+        }
+        this.aliases[pluginName] = pluginName;
+      }
+
+      aliases.forEach((alias: string) => {
+        if (this.aliases[alias] !== undefined) {
+          throw new Error(
+            `Duplicate alias detected: ${pluginName} is trying to claim ${alias}, but ${this.aliases[alias]} has already claimed it.`
+          );
+        }
```

Comment:
This will fail to add aliases after the first duplicate is found.

Reply:
Joey wants it to error on finding a duplicate plugin so that is meant to happen.

Reply:
it doesn't make sense. [alias1, duplicatedAlias, alias2]. alias2 would never be added.

Reply:
it doesn't make sense. [alias1, duplicatedAlias, alias2]. alias2 would never be added.

Reply:
Do you mean if a plugin has something like alias = [cats, cats], or if the Cat plugin had alias = [cat] then it will crash?  Cause the idea of the error was meant to stop a second author from trying to overwrite it.  If that is the issue I can do:

`(this.aliases[alias] !== undefined && this.aliases[alias] !== 'pluginName')`

Reply:
no the issue is that the behaviour should be "if any alias is invalid, dont add any of them, let the developer know they messed up" __not__ "keep adding aliases until one is invalid, then dont add anymore"

Reply:
@c650 i think you are misunderstanding. the behavior with how duplicate plugin names work is if the bot is registering one and comes across a duplicate key, we are stopping execution of program so that the developer is able to notice that something went wrong.

i was suggesting to apply the same logic here to remain consistent with what we've done in the past. i would prefer to not have duplicate plugin to stop execution of program and then duplicate alias to throw a log in the console. it makes sense to remain consistent with how we're handling the duplicate case. this decision is still up for debate but it should be dealt with in another pr if we want to change it. for now i am for stopping execution of program if duplicate alias is found